### DISCUSSION TITLE
Consult-buffer "not "remembers" open buffers from previous sessions.

### DISCUSSION THREAD

**Discussion by gety9** [minad/consult#931] [Q&A]
Title.

Any time i kill-emacs, and then restart it starts from message & scratch - nothing else. (Only the 2 iteams are shown in M-x consult-buffer)

Do i need to do anything else besides enabling savehist?
M-x "remembers" (commands from previous sessions are shown on top)

History file is updated. (there are savehist-minibuffer-history-variables, file-name-history, consult--buffer-history, etc)

But consult--buffer-history has only 3 items. (i opened over sessions much more)

What can it be? How to trouble shoot?

---------
Parts of config that i think maybe related to the issues:
```
(setq make-backup-files nil)
(setq auto-save-file-name-transforms   `(("\\`/[^/]*:\\([^/]*/\\)*\\([^/]*\\)\\'" ,(concat user-emacs-directory "\\2") t)))
(setq lock-file-name-transforms        `(("\\`/[^/]*:\\([^/]*/\\)*\\([^/]*\\)\\'" ,(concat user-emacs-directory "\\2") t)))
(use-package savehist
  :demand t
  :init
  (savehist-mode t))
```

```
(use-package consult
  :config
  (consult-customize
    consult-buffer :preview-key "k"
    consult-line   :preview-key '(:debounce 0.5 any)
  ))

(use-package embark
  :demand t)
(use-package embark-consult)

(use-package vertico
  :init
  (vertico-mode))

(use-package marginalia
  :init
  (marginalia-mode))

(use-package orderless
  :demand t
  :custom
  (completion-styles '(orderless))
  (  read-file-name-completion-ignore-case t
     read-buffer-completion-ignore-case t
     completion-ignore-case t))
```

**Comment by gety9** [comment]
I start emacs with init dir param (`emacs --init-directory ~/.emacs.d2`)

Not sure if that can be causing issue, but why that M-x work fine? (remembers commands from previous sessions...)

**Reply by minad** [reply]
Did you compare `consult-buffer` with `switch-to-buffer`? Do you see a difference? You probably also want to enable `recentf-mode` to remember recently opened files.

**Reply by gety9** [reply]
@minad 

-, switch-to-buffer also doesn't remember buffers. On restart it knows about 2 (but shows only 1 - "messages" - cause "scratch is already current buffer")

-, "recentf-mode" i will try that. But is it required for buffers to work? Or is "savehist" supposed to be enough? (I want to understand if something is wrong with my config (currently) before adding new packages (recentf, etc).)

**Reply by gety9** [reply]
@minad 

Seems like savehist is only for commands, search, etc (that's why M-x remembers history)
And for buffers I need to enable desktop-save-mode, for files recentf-mode.

Could you please advise on optimal workflow when managing buffers / files?

Say I use consult-buffer, should I use groups? Or bookmarks? Or tabs? to organize them / switch between them? Or should I use projects?

So many options - I am lost :)

**Comment by minad** [comment]
gety9 ***@***.***> writes:

> Could you please advise on optimal workflow when managing buffers / files?

My recommendation is to enable savehist-mode and recentf-mode. I also
suggest to use bookmarks, where applicable. If you use version control,
project usage will come naturally.
